### PR TITLE
correct mappings

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

File: ugrid/ugrid.py [hunk 1]
```diff
@@ -45,13 +46,16 @@ def __init__(self, file_path, method):
...
+        lib_path = Path(__file__).parent
         if system == "Windows":
-            lib_path = Path(__file__).parent / "UGridApi.dll"
+            lib_path = os.path.join(lib_path, "UGridApi.dll")
         elif system == "Linux":
-            lib_path = Path(__file__).parent / "UGridApi.so"
+            lib_path = os.path.join(lib_path, "UGridApi.so")
         elif system == "Darwin":
-            lib_path = Path(__file__).parent / "UGridApi.dylib"
+            lib_path = os.path.join(lib_path, "UGridApi.dylib")
```
Comment (ahmad-el-sayed) [inline ugrid/ugrid.py:55]: Where is the macOS build config?

File: ugrid/ugrid.py [hunk 2]
```diff
@@ -45,13 +46,16 @@ def __init__(self, file_path, method):
 
         # Determine OS
         system = platform.system()
+        lib_path = Path(__file__).parent
         if system == "Windows":
-            lib_path = Path(__file__).parent / "UGridApi.dll"
+            lib_path = os.path.join(lib_path, "UGridApi.dll")
         elif system == "Linux":
-            lib_path = Path(__file__).parent / "UGridApi.so"
+            lib_path = os.path.join(lib_path, "UGridApi.so")
```
Comment (ahmad-el-sayed) [inline ugrid/ugrid.py:53]: TC job can be made a bit nicer with subprojects.

File: ugrid/ugrid.py [hunk 3]
```diff
@@ -45,13 +46,16 @@ def __init__(self, file_path, method):
...
         elif system == "Darwin":
-            lib_path = Path(__file__).parent / "UGridApi.dylib"
+            lib_path = os.path.join(lib_path, "UGridApi.dylib")
         else:
+            if not system:
+                system = "Unknown OS"
             raise OSError(f"Unsupported operating system: {system}")
```
Comment (ahmad-el-sayed) [inline ugrid/ugrid.py:59]: Can be moved to a free function.

File: tests/test_mesh2d.py
Comment (ahmad-el-sayed) [inline tests/test_mesh2d.py:1]: Shall we add a formatting action?

File: ugrid/c_structures.py [hunk 4]
```diff
@@ -33,9 +33,9 @@ def pad_and_join_list_of_strings(string_list: list, str_size: int):
 
 def numpy_array_to_ctypes(arr):
     """Cast an array to ctypes only if its len is not 0"""
-    if len(arr) == 0:
-        return None
-    return as_ctypes(arr)
+    if arr is not None and len(arr) > 0:
```
Comment (ahmad-el-sayed) [inline ugrid/c_structures.py:36]: Can simply use `if not arr`

File: ugrid/ugrid.py (hunk 1 above)
Comment (lucacarniato) [reply ugrid/ugrid.py:55]: now only windows is supported

File: ugrid/ugrid.py (hunk 3 above)
Comment (lucacarniato) [reply ugrid/ugrid.py:59]: moved to a private function

File: tests/test_mesh2d.py
Comment (lucacarniato) [reply tests/test_mesh2d.py:1]: yes

File: ugrid/c_structures.py (hunk 4 above)
Comment (lucacarniato) [reply ugrid/c_structures.py:36]: No, you get an error

**Pull request merged**